pkg/controllers/updaterun/initialization_integration_test.go

```diff
@@ -0,0 +1,758 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package updaterun
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/utils"
+)
+
+var (
+	cmpOptions = []cmp.Option{
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+		cmpopts.IgnoreFields(metav1.Condition{}, "Message"),
+	}
+)
+
+var _ = Describe("Updaterun initialization tests", func() {
+	var updateRun *placementv1alpha1.ClusterStagedUpdateRun
+	var crp *placementv1beta1.ClusterResourcePlacement
+	var policySnapshot *placementv1beta1.ClusterSchedulingPolicySnapshot
+	var updateStrategy *placementv1alpha1.ClusterStagedUpdateStrategy
+	var resourceBindings []*placementv1beta1.ClusterResourceBinding
+	var targetClusters []*clusterv1beta1.MemberCluster
+	var unscheduledCluster []*clusterv1beta1.MemberCluster
+	var resourceSnapshot *placementv1beta1.ClusterResourceSnapshot
+	var clusterResourceOverride *placementv1alpha1.ClusterResourceOverrideSnapshot
+
+	BeforeEach(func() {
+		testUpdateRunName = "updaterun-" + utils.RandStr()
+		testCRPName = "crp-" + utils.RandStr()
+		testResourceSnapshotName = "snapshot-" + utils.RandStr()
+		testUpdateStrategyName = "updatestrategy-" + utils.RandStr()
+		testCROName = "cro-" + utils.RandStr()
+		updateRunNamespacedName = types.NamespacedName{Name: testUpdateRunName}
+
+		updateRun = getTestClusterStagedUpdateRun()
+		crp = getTestClusterResourcePlacement()
+		policySnapshot = getTestClusterSchedulingPolicySnapshot(1)
+		updateStrategy = getTestClusterStagedUpdateStrategy()
+		clusterResourceOverride = getTestClusterResourceOverride()
+
```

Comment:
Are those "create" functions or "get" functions?

Reply:
I changed to `generate-` prefix instead.